src/org/thoughtcrime/securesms/database/MmsDatabase.java

```diff
@@ -413,6 +414,57 @@ public void markAsNotified(long id) {
     database.update(TABLE_NAME, contentValues, ID_WHERE, new String[] {String.valueOf(id)});
   }
 
+  public void markMessagesAsPinned(long threadId, long messageId){
+    markMessagesAsPinned(THREAD_ID + " = ? AND " + ID + " = ? AND "+ PINNED + " = 0", new String[] {String.valueOf(threadId), String.valueOf(messageId)});
+  }
+
+  public void markMessagesAsPinned(String where, String[] args) {
```

Comment:
Should be a private method, so that only the previous markMessagesAsPinned is exposed through the interface